server.js

```diff
@@ -23,50 +23,48 @@ app.use(cookieParser());
 
 // get requests
 app.get("/", async (req, res) => {
-    const { cookies } = req
     try {
-        const currentSession = await Session.findById(cookies.session_id)
-        if (currentSession === null) {
-            res.render('index', currentSession);
-        } else {
-            res.render('index', currentSession);
+        if (!req.cookies.session) {
+            res.render('index');
+            return;
         }
-
+        console.log(req.cookies.session);
+        const currentSession = await Session.findById(req.cookies.session);
+        if (!currentSession) {
+            res.clearCookie('session');
+            res.render('index');
+            return;
+        }
```

Comment:
Here you did it correctly, you check to see if the session you got is even valid.